Makefile

```diff
@@ -32,6 +33,32 @@ doc:
 	cp ocaml/doc/*.dot ocaml/doc/doc-convert.sh $(XAPIDOC)
 	find ocaml/doc -name "*.md" -not -name "README.md" -exec cp {} $(XAPIDOC)/markdown/ \;
 
+sdk:
+	cp $(SHAREDIR)/sm/XE_SR_ERRORCODES.xml ocaml/sdk-gen/csharp/XE_SR_ERRORCODES.xml
+	dune build --profile=$(PROFILE) \
+		ocaml/sdk-gen/c/gen_c_binding.exe \
+		ocaml/sdk-gen/csharp/gen_csharp_binding.exe \
+		ocaml/sdk-gen/java/main.exe \
+		ocaml/sdk-gen/powershell/gen_powershell_binding.exe
+	dune build --profile=$(PROFILE) -f\
+		@ocaml/sdk-gen/c/generate \
+		@ocaml/sdk-gen/csharp/generate \
+		@ocaml/sdk-gen/java/generate \
+		@ocaml/sdk-gen/powershell/generate
+	mkdir -p $(XAPISDK)/c
+	mkdir -p $(XAPISDK)/csharp
+	mkdir -p $(XAPISDK)/java
+	mkdir -p $(XAPISDK)/powershell
+	mkdir -p $(XAPISDK)/python/samples
+	cp -r _build/default/ocaml/sdk-gen/c/autogen/* $(XAPISDK)/c
+	cp -r _build/default/ocaml/sdk-gen/csharp/autogen/* $(XAPISDK)/csharp
+	cp -r _build/default/ocaml/sdk-gen/java/autogen/* $(XAPISDK)/java
+	cp -r _build/default/ocaml/sdk-gen/powershell/autogen/* $(XAPISDK)/powershell
+	cp -r ocaml/sdk-gen/python/samples/*.py $(XAPISDK)/python/samples
+	cp scripts/examples/python/XenAPI/XenAPI.py $(XAPISDK)/python
+	sh ocaml/sdk-gen/windows-line-endings.sh $(XAPISDK)/csharp
+	sh ocaml/sdk-gen/windows-line-endings.sh $(XAPISDK)/powershell
```

Comment:
This looks like something we eventually could get dune to do? I should be possible to get dune to automatically put the installation targets into the right location under `_build/install`.

Reply:
The install stanza does not seem to support multiple files/directories yet, this issue is still open: https://github.com/ocaml/dune/issues/1868. I also tried the sites stanza, but it needs a higher dune version. Any other suggestions?

Reply:
Hmm... I don't think its the install stanza you need here, which is for installing files into the wider filesystem. The build command puts the build targets under `_build/install`. So I think these files need to be marked as targets somehow (I'm probably getting the terminology wrong). I'll take a look as well.

Reply:
Oh, I see now that it is indeed the install "stanza" that you need, which is implicit when your use `public_name` in `executable` and things like that.

Reply:
Anyway, let's leave it like this for now (we'll have something similar for `make doc`).